clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -1,34 +1,69 @@
 #![cfg(feature = "unstable-dynamic")]
 
+macro_rules! assert_completion {
```

Comment:
This feels too complicated of a macro, both in API and in implementation.

If there is something specific we need from a macro, we should at least delegate the code logic of it to a function.

Deconstructing not-quite arrays is also a complicated API for it